wordpress-mobile/WordPress-Android
# Move editor preference option to Site Settings screen

Move the "default editor for new posts" toggle from App Settings screen to Site Settings screen.

Details here: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1233

**Test 1** - Account already migrated to version 12.9 and that has seen the auto-enable dialog
(the app has the global option set to TRUE)
- Open app settings and see there is no editor option
- Go to site settings. The default editor preference is there and it should be `TRUE`
- Change it to false
- Re-open settings, it should show the correct value
- Tap to create a new post, the right editor (Aztec) should be shown

**Test 2** - Account already migrated to version 12.9 that has not yet seen the auto-enable dialog
_(so basically an account that didn't open a GB based post in v12.9)_
- Open app settings and see there is no editor option
- Go to site settings. The default editor preference is there and it should be `false`
- Tap to create a new post, the right editor (Aztec) should be shown
- Open a blocks based post 
- The dialog that informs of the auto-enable should appear 
- Re-open settings, it should show the correct value (`true`)
- Tap to create a new post, the right editor (GB) should be shown.

**Test 3** - Account already migrated to version 12.9 that has set the editor option to `FALSE` after the auto-opt in dialog.
_(so basically an account that opened a GB based post in v12.9, but turned it OFF later)_
- Open app settings and see there is no editor option
- Go to site settings. The default editor preference is there and it should be `false`
- Tap to create a new post, the right editor (Aztec) should be shown
- Open a blocks based post 
- NO dialog that informs of the auto-enable should appear 
- Re-open settings, it should show the correct value (`false`)
- Tap to create a new post, the right editor (Aztec) should be shown.

**Test 4** - NEW account
- Open app settings and see there is no editor option
- Go to site settings. The default editor preference is there and it should be `true`
- Tap to create a new post, the right editor (GB) should be shown
- Go to site settings and change it to false
- Re-open settings, it should show the correct value
- Tap to create a new post, the right editor (Aztec) should be shown.

**Test 5** - Self hosted sites - Upgrading from 12.9 where the block editor was enabled globally
- Open app settings and see there is no editor option
- Go to site settings. The default editor preference is there and it should be `true`
- Tap to create a new post, the right editor (GB) should be shown
- Go to site settings and change it to false
- Re-open settings, it should show the correct value
- Tap to create a new post, the right editor (Aztec) should be shown

**Test 6** - Self hosted sites - new install
- Open app settings and see there is no editor option
- Go to site settings. The default editor preference is there and it should be `false`
- Tap to create a new post, the right editor (Aztec) should be shown
- Go to site settings and change it to true
- Re-open settings, it should show the correct value
- Tap to create a new post, the right editor (GB) should be shown

Update release notes:

- [ x ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.
